src/main/java/seedu/address/model/reminder/UniqueReminderList.java

```diff
@@ -0,0 +1,140 @@
+
+    /**
+     * Constructor of UniqueReminderList
+     */
+    private UniqueReminderList() {
+        internalList = FXCollections.observableArrayList();
+        internalUnmodifiableList = FXCollections.unmodifiableObservableList(internalList);
+        personToReminderMap = new HashMap<>();
+    }
+
+    /**
+     * Returns true if the list contains an equivalent Reminder as the given argument.
+     */
+    public boolean contains(Reminder toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::equals);
+    }
+
+    /**
+     * Factory method of UniqueReminderList
+     *
+     * @return Produced UniqueReminderList object that is a Singleton.
+     */
+    public static UniqueReminderList getInstance() {
+        if (reminderList == null) {
+            reminderList = new UniqueReminderList();
+        }
+        return reminderList;
+    }
+
+    /**
+     * Adds a Reminder to the list.
+     * The Reminder must not already exist in the list.
+     */
+    public void add(Person person, Interaction interaction) {
+        requireAllNonNull(person, interaction);
+        Reminder toAdd = new Reminder(person.getName(), person.getPhone(), person.getTags(), person.getLead(),
+                interaction.getDate());
+        if (contains(toAdd)) {
+            throw new DuplicateReminderException();
+        }
+
+        if (personToReminderMap.containsKey(person)) {
+            internalList.remove(personToReminderMap.get(person));
+        }
+
+        internalList.add(toAdd);
+        personToReminderMap.put(person, toAdd);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Reminder> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    /**
+     * Returns the list of reminders after a specific date.
+     *
+     * @param date The date that which is used to retrieve the list of reminders.
+     * @return the list of reminders mapped from the given date.
+     */
+    public ObservableList<Reminder> getRemindersAfterDate(Date date) {
+        ObservableList<Reminder> reminderList = FXCollections.observableArrayList();
+        List<Reminder> retrievedReminders = personToReminderMap.entrySet()
+                .stream()
+                .map(x -> x.getValue())
+                .filter(a->a.compareTo(date) == 1)
+                .collect(Collectors.toList());
+        if (retrievedReminders == null) {
+            return reminderList;
+        }
+        reminderList.addAll(retrievedReminders);
+        return reminderList;
+    }
+
+    /**
+     * Resets the internal list of reminders using the given list of persons.
+     *
+     * @param personList The persons from which the reminders are to be produced.
+     */
+    public void setReminders(ObservableList<Person> personList) {
+        for (Person person : personList) {
```

Comment:
Should we clear the reminder list first? This will be useful if any `Person` has been deleted.